Product/Production/Services/DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/docsubmission/outbound/StandardOutboundDocSubmission.java

```diff
@@ -165,39 +172,42 @@ private RegistryResponseType getResponseFromTarget(
     }
 
     private gov.hhs.fha.nhinc.common.nhinccommonproxy.RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType createRequestForNhin(
-            RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType request) {
-
-        gov.hhs.fha.nhinc.common.nhinccommonproxy.RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType nhinRequest =
-            new gov.hhs.fha.nhinc.common.nhinccommonproxy.RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType();
+        RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType request) {
 
-        NhinTargetSystemType targetSystemType = new NhinTargetSystemType();
-        targetSystemType.setHomeCommunity(getNhinTargetHomeCommunity(request));
+        gov.hhs.fha.nhinc.common.nhinccommonproxy.RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType nhinRequest
+            = new gov.hhs.fha.nhinc.common.nhinccommonproxy.RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType();
 
-        nhinRequest.setNhinTargetSystem(targetSystemType);
+        nhinRequest.setNhinTargetSystem(getTargetSystemTypeFromCommunity(getNhinTargetHomeCommunity(request)));
```

Comment:
The following code can be used to create this NhinTargetSystemType from NhinTargetCommunitiesType extracted from request. This would reduce 3 methods from this class.
NhinTargetSystemType target = getMessageGeneratorUtils().convertFirstToNhinTargetSystemType(targets);